### PR TITLE
Optimize WindowedValueCoder

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @vikkyrk

File: sdks/python/apache_beam/coders/coder_impl.py [hunk 1]
```diff
@@ -514,19 +514,28 @@ class WindowedValueCoderImpl(StreamCoderImpl):
...
 
   def encode_to_stream(self, value, out, nested):
-    self._value_coder.encode_to_stream(value.value, out, True)
-    self._timestamp_coder.encode_to_stream(value.timestamp, out, True)
-    self._windows_coder.encode_to_stream(value.windows, out, True)
+    wv = value  # type cast
+    self._value_coder.encode_to_stream(wv.value, out, True)
+    if isinstance(self._timestamp_coder, TimestampCoderImpl):
+      # Avoid creation of Timestamp object.
+      out.write_bigendian_int64(wv.timestamp_micros)
```
Comment (vikkyrk) [inline sdks/python/apache_beam/coders/coder_impl.py]: TimestampCoderImpl does the same, so we are not creating a timestamp object if we just do timestamp_coder.encode_to_stream?

**Comment by vikkyrk** [comment]
Minor comments, but LGTM otherwise

File: sdks/python/apache_beam/coders/coder_impl.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.py]: wv.timestamp creates the timestamp object. Here we just use the micros field directly.

File: sdks/python/apache_beam/coders/coder_impl.pxd [hunk 2]
```diff
@@ -137,3 +138,6 @@ cdef class WindowedValueCoderImpl(StreamCoderImpl):
 
   @cython.locals(c=CoderImpl)
   cpdef get_estimated_size_and_observables(self, value, bint nested=?)
+
+  @cython.locals(wv=windowed_value.WindowedValue)
+  cpdef encode_to_stream(self, value, OutputStream stream, bint nested)
```
Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.pxd]: This code is only to declare the type of wv (which we're doing here in the pxd file to keep Cython optional).

Comment (vikkyrk) [inline sdks/python/apache_beam/coders/coder_impl.pxd]: How about cython.returns decorator to declare the return type for decode_from_stream. Just thinking out loud, as this is new to me.

Comment (robertwb) [inline sdks/python/apache_beam/coders/coder_impl.pxd]: We could. We'd have to specify it for the supertype. In general, however, I try to specify as little as needed.

**Pull request closed without merging**